src/app/components/itinerary-details/itinerary-details.component.ts

```diff
@@ -31,26 +28,31 @@ export class ItineraryDetailsComponent {
 
   selectedTrip$ = this.store.select(selectSelectedTrip);
   loading$ = this.store.select(selectLoadingState);
+  exchangeRates$ = this.store.select(selectExchangeRates);
 
   tripId = signal<string>(this.activatedRoute.snapshot.params['id']);
```

Comment:
Consider storing the tripId in the state management, that way you can simply derive it from there instead of grabbing it from the url. It would help prevent people from going straight to the page with random IDs typed in

Reply:
Tim mentioned having deterministic routes provides a better user experience. In the case they reload it is quick to know what trip you were on based on the route. There are safety checks in place in case the trip does not exist (i.e. the user entered random letters in the url) to route the user to home.